objects/src/accounts/component/template/storage/mod.rs

```diff
@@ -0,0 +1,585 @@
+    Map {
+        /// The human-readable name of the map slot.
+        name: String,
+        /// An optional description for the slot, explaining its purpose.
+        description: Option<String>,
+        /// The numeric index of this map slot in the component's storage.
+        slot: u8,
+        /// A list of key-value pairs to initialize in this map slot.
+        map_entries: Vec<MapEntry>,
+    },
+
+    /// A multi-slot entry, representing a single logical value across multiple slots.
+    MultiSlot {
+        /// The human-readable name of this multi-slot entry.
+        name: String,
+        /// An optional description for the slot, explaining its purpose.
+        description: Option<String>,
+        /// The indices of the slots that form this multi-slot entry.
+        slots: Vec<u8>,
+        /// A list of values to fill the logical slot, with a length equal to the amount of slots.
+        values: Vec<WordRepresentation>,
+    },
+}
+
+impl StorageEntry {
+    /// Creates a new [`StorageEntry::Value`] variant.
+    pub fn new_value(
+        name: impl Into<String>,
+        description: Option<impl Into<String>>,
+        slot: u8,
+        value: impl Into<WordRepresentation>,
+    ) -> Self {
+        StorageEntry::Value {
+            name: name.into(),
+            description: description.map(Into::<String>::into),
+            slot,
+            value: value.into(),
+        }
+    }
+
+    /// Creates a new [`StorageEntry::Map`] variant.
+    pub fn new_map(
+        name: impl Into<String>,
+        description: Option<impl Into<String>>,
+        slot: u8,
+        map_entries: Vec<MapEntry>,
+    ) -> Self {
+        StorageEntry::Map {
+            name: name.into(),
+            description: description.map(Into::<String>::into),
+            slot,
+            map_entries,
+        }
+    }
+
+    /// Creates a new [`StorageEntry::MultiSlot`] variant.
+    pub fn new_multi_slot(
+        name: impl Into<String>,
+        description: Option<impl Into<String>>,
+        slots: Vec<u8>,
+        values: Vec<impl Into<WordRepresentation>>,
+    ) -> Result<Self, AccountComponentTemplateError> {
+        if slots.len() != values.len() {
+            return Err(AccountComponentTemplateError::MultiSlotArityMismatch);
+        }
+
+        for window in slots.windows(2) {
+            if window[1] != window[0] + 1 {
+                return Err(AccountComponentTemplateError::NonContiguousSlots(
+                    window[0], window[1],
+                ));
+            }
+        }
+
+        Ok(StorageEntry::MultiSlot {
+            name: name.into(),
+            description: description.map(Into::<String>::into),
+            slots,
+            values: values.into_iter().map(Into::into).collect(),
+        })
+    }
+
+    /// Returns the slot indices that the storage entry covers.
+    pub fn slot_indices(&self) -> &[u8] {
```

Comment:
Similar to the previous comment, this could also be a `Range`.

Reply:
Created https://github.com/0xPolygonMiden/miden-base/issues/1052 for follow-ups